clients/go/test/api_uploads_test.go

```diff
@@ -17,34 +17,52 @@ import (
 	"testing"
 
 	"github.com/antihax/optional"
-	"github.com/phrase/phrase-go"
+	"github.com/phrase/phrase-go/v2"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func Test_phrase_UploadsApiService(t *testing.T) {
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// Send the mock response
-		response := `{"id": "1", "filename": "test.json", "format": "json", "state": "valid" }`
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
-
-		w.Write([]byte(response))
-	}))
-
-	defer server.Close()
-
-	configuration := phrase.NewConfiguration()
-	configuration.BasePath = server.URL
-	apiClient := phrase.NewAPIClient(configuration)
-
 	t.Run("Test UploadsApiService UploadCreate", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// assert request body for proper serialization of tested format_options
+			r.ParseMultipartForm(0)
+			assert.Equal(t, r.FormValue("format_options[translation_columns][en]"), "f")
+			assert.Equal(t, r.FormValue("format_options[translation_columns][de]"), "e")
+			assert.Equal(t, r.FormValue("format_options[top_level_key]"), "100")
+
+			// Send the mock response
+			response := `{"id": "1", "filename": "test.json", "format": "json", "state": "valid" }`
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+	
+			w.Write([]byte(response))
+		}))
+	
+		defer server.Close()
+	
+		configuration := phrase.NewConfiguration()
+		configuration.BasePath = server.URL
+		apiClient := phrase.NewAPIClient(configuration)
+
 		file, _ := os.Create("testfile.json")
 		fileFormat := optional.NewString("json")
 		fileObject := optional.NewInterface(file)
 		localeId := optional.NewString("99")
 
-		localVarOptionals := phrase.UploadCreateOpts{FileFormat: fileFormat, File: fileObject, LocaleId: localeId }
+		// setting format_options
+		formatOptions := make(map[string]interface{})
+		nestedOptions := make(map[string]interface{})
```

Comment:
I have assumed, the inner map, if provided will be also of type `map[string]interface{}`. Allowing for values of different types.